serif_shared/src/commonMain/kotlin/xyz/room409/serif/serif_shared/MatrixClient.kt

```diff
@@ -106,9 +110,38 @@ class MatrixClient {
                     body = LoginRequest(username, password)
                 }
             }
-            return Success(MatrixSession(client, loginResponse.access_token))
+
+            //Save to DB
+            println("Saving session to db")
+            val new_transactionId : Long = 0
+            Database.saveSession(username, loginResponse.access_token, new_transactionId)
+
+            return Success(MatrixSession(client, loginResponse.access_token, new_transactionId))
         } catch (e: Exception) {
             return Error("Login failed", e)
         }
     }
+    fun loginFromSavedSession(username: String): Outcome<MatrixSession> {
+        val client = HttpClient() {
+            install(JsonFeature) {
+                serializer = KotlinxSerializer(kotlinx.serialization.json.Json {
+                    ignoreUnknownKeys = true
+                })
+            }
+        }
+        //Load from DB
+        println("loading specific session from db")
+        val sessions = Database.getStoredSessions()
+        for((user, tok, transactionId) in sessions) {
+            if(user == username) {
+                return Success(MatrixSession(client, tok, transactionId))
+            }
+        }
+        return Error("No Saved Session for $username")
+    }
+
+    fun getStoredSessions() : List<String> {
```

Comment:
Should this be a property instead of a getter? Might be part of a larger style discussion